.github/workflows/main.yml

```diff
@@ -1,15 +1,21 @@
 name: CI
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - master
+  pull_request:
 
 jobs:
   build:
-
     runs-on: ubuntu-latest
 
     steps:
     - uses: actions/checkout@v2.3.4
       with:
+        # Checkout the PR HEAD commit instead of merge commit
+        ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
I no longer remember why I did this.

Reply:
Ok, now I remember why I did this - this is because we tag the executor image with the current git hash. If we used the test merge commit, it'd be tagged with a SHA that doesn't correspond to any actual commit, so if we tried to deploy the branch somewhere, there wouldn't be an image corresponding to the event present in the image registry.

However, I'm pretty confident that this will actually fail once we're on master, as it seems as though this property is only available with the `pull_request` target, not for `push`. I'm thinking an alternative would be to provide the value of `github.event.pull_request.head.sha` to `tools/executor/{build,release}.js` via an environment variable, and default to picking it up via `git rev-parse HEAD` if an environment variable is not provided? Need to noodle on this some more.